recipe/build.sh

```diff
@@ -1,13 +1,77 @@
 #!/bin/bash
 
+set -ex
+
 # Compile
 # -------
 chmod +x configure
 
+# Frustratingly, Qt's configure checks do not honor the standard build
+# environment variables ($CC, $*FLAGS, etc) in a very consistent way. In
+# particular, they are largely ignored during configuration. AFAICT, the best
+# way to get them to apply during both configuration and the primary build is
+# to modify the "mkspec" files used in the build. (You can pass #defines to
+# the configuration script, but you can't pass linker flags like
+# `-Wl,-rpath`.)
+#
+# Aside: our toolchain environments unfortunately mix preprocessor and
+# non-preprocessor flags in both $CFLAGS and $CPPFLAGS.
+
 if [ $(uname) == Linux ]; then
+    compiler_mkspec=qtbase/mkspecs/common/g++-base.conf
+    flag_mkspec=qtbase/mkspecs/linux-g++/qmake.conf
+
+    # The Anaconda gcc7 compiler flags specify -std=c++17 by default, which
+    # activates features that break compilation. Begone!
+    CXXFLAGS=$(echo $CXXFLAGS | sed -E 's@\-std=[^ ]+@@')
+    export CXXFLAGS="$CXXFLAGS -std=c++11"
+
+    # This warning causes a huge amount of spew in the build logs.
+    if [ "$cxx_compiler" = gxx ] ; then
+        CXXFLAGS="$CXXFLAGS -Wno-expansion-to-defined"
+    fi
+
+    export LDFLAGS="$LDFLAGS -Wl,-rpath-link,$PREFIX/lib"
+    export CPPFLAGS="$CPPFLAGS -DXK_dead_currency=0xfe6f -DXK_ISO_Level5_Lock=0xfe13"
+    export CPPFLAGS="$CPPFLAGS -DFC_WEIGHT_EXTRABLACK=215 -DFC_WEIGHT_ULTRABLACK=FC_WEIGHT_EXTRABLACK"
+    export CPPFLAGS="$CPPFLAGS -DGLX_GLXEXT_PROTOTYPES"
+else
+    compiler_mkspec=qtbase/mkspecs/common/clang.conf
+    flag_mkspec=qtbase/mkspecs/macx-clang/qmake.conf
+
+    export LDFLAGS="$LDFLAGS -Wl,-rpath,$PREFIX/lib"
+    export CXXFLAGS="$CXXFLAGS -std=c++11"
```

Comment:
This is a significant divergence from `defaults`. Why not leave Qt to set its own flags?

Reply:
Well, I guess my overall philosophy is, if you're trying to use a consistent set of compiler flags in your ecosystem and that ecosystem contains a complex, widely-used C++ library, having that library ignore your flags and do its own thing is risky from an ABI standpoint.

But if `defaults` has a working recipe it does seem wise to hew closely to it. I was just building off of what we had here. The choice of system/CDT vs. C-F X.org X11 libraries could make a difference, but I agree that ideally the PR here wouldn't change the status quo. A related factor was that I don't know how to get a C-F recipe to use CDT packages, although I must admit I didn't actually try to figure that out.

Reply:
There's no great complexity to using CDT packages, you just include them and they should work, in theory. It's also really easy to create and build your own (someone needed libusb recently and did this). Very occasionally you run into header conflicts between new stuff and old stuff but I've not run into anything too difficult to solve or work-around.

Reply:
@pkgw, `-std=c++17` is already set in the new compiler. Can you do this only for toolchain build? Also you can remove the line above (LDFLAGS) as it's done by both toolchain and anaconda compilers.

Reply:
@conda-forge/qt, I think a decision needs to be done about this. Should we proceed with the Xorg packages or should we try to use the CDT ones?

My vote is for using the Xorg ones given that a) @pkgw has already make Qt compile with them; and b) I guess very few people has time to start creating the CDT ones.